### ISSUE TITLE
add performance metric(s) for seasonal effect of fishing in one country on the other

### ISSUE THREAD

**Issue by kristinmarshall-NOAA** [pacific-hake/pacifichakemse#6]
Related to the attainment scenarios, there was interest in us looking at the effects of fishing in one area on the availability of biomass in the other area in the subsequent season.  For example, does catch removed in US waters in spring affect the available biomass to CA fisheries in summer?  Or, does fishing in Canadian waters in summer affect the availability of fish to US fisheries in fall?

To look at this, I think we need some new metrics.  One idea I have is to summarize spatial and seasonal vulnerable biomass/quota as the probability that biomass/quota is less than 1 (or some lower threshold value) in Canada in spring and summer, and in the US in summer and fall.  And then we might also need to condition that on the other country achieving their quota in the previous season-I'm not sure.

It looks like @cgrandin has added a [warning message](https://github.com/pacific-hake/pacifichakemse/blob/5871349b7af0159afdc2d4fe4f0378ff48daccd5/R/run_season_loop_om.R#L133) when the available biomass is 0.9 or less than what the allowed catch would remove in a given area and season, so maybe adding a counter inside this seasonal loop would be one way to do it.  Or, perhaps we can get the same from comparing catch, quota, and biomass in the outputs already being generated.

What do you all think?

**Comment by aaronmberger-nwfsc** [comment]
My initial thought is that this seems logical.  Questions would be (among others perhaps):
- how to divide the quota up by season?  Would this be based on historical seasonal catch proportions?
- alternatively/additionally we could use a cumulative metric/plot over seasons in a year such that the quota in the seasonal calculations is just the annual quota in order to see how the quota gets "filled" during the year and where it ends up at the end of the year.  Adjacent seasonal comparisons between countries could be evaluated according to how "full" their quota is.  This would of course all be related to seasonal movement assumed in the OM so we would want to relate any results to that set of assumptions.

I'm a bit skeptical that comparing adjacent seasons by country will tell us much, but that's just an initial feeling.  It just is going to be hard to interpret, especially when there is some indication that there are other factors affecting ability to catch quota (attainment) such as market factors and perhaps even catch capacity.   For example, see figures below of information taken straight from the assessment document - looking at percent [annual] attainment relative to beginning of the year SSB (and relative to age 2+ below those).  There is almost no relationship for US fishery and a negative relationship for Canada for SSB, and as expected that relationship lessens for Canada when looking at age 2+.  This could be due to flexibilities in markets, fishing capacity, and/or a host of other factors besides or in addition to a simple metric that tracks whether country-specific attainment is related (without accounting for all these other factors).  We don't do these types of analyses using fishery-dependent CPUE data for similar reasons.  Sorry - just playing devils advocate a bit here.  I will think more about this as I only have a few minutes now (technically no holiday).

![image](https://user-images.githubusercontent.com/7946234/100375501-bca4a600-2fc2-11eb-99ec-41aa8aa0df11.png)

![image](https://user-images.githubusercontent.com/7946234/100375515-c1695a00-2fc2-11eb-8671-0ba530145011.png)

![image](https://user-images.githubusercontent.com/7946234/100375606-e8c02700-2fc2-11eb-9f79-154a4e310b84.png)

![image](https://user-images.githubusercontent.com/7946234/100375630-f675ac80-2fc2-11eb-96a0-5696dd7eb13e.png)

**Comment by cgrandin** [comment]
Thanks for doing those plots Aaron. I share your concerns about how informative output of such an analysis would be, markets have affected catch a great deal in the last decade and it's something we aren't modeling.

Instead of worrying about quota by season maybe we could just look at cumulative catch patterns and assume the total catch at the end of the year as the attained proportion we want to model. I think this is the first step in Aaron's idea in his second point above.

Here's some plots we show at the JTC meeting every year showing cumulative catch by sector for the countries.

In recent years in Canada, the cumulative catch taken is roughly linear from May 1 to November 1 - Freezer Trawlers:
![CumulativeCatchFT](https://user-images.githubusercontent.com/4551051/100387253-7fe5a880-2fdc-11eb-9eb5-b189b04add11.png)

Canadian Shoreside:
![CumulativeCatchSS](https://user-images.githubusercontent.com/4551051/100387448-f97d9680-2fdc-11eb-9b44-9cfc7b67c118.png)

In the US, only the Shorside has this linear cumulative catch:

![image](https://user-images.githubusercontent.com/4551051/100387590-4fead500-2fdd-11eb-89a8-8a0f08be194b.png)
but the Catcher/processor and mothership sectors have flat spots in June-July which is when they go to Alaska for Pollock.

Catcher/Processor:
![image](https://user-images.githubusercontent.com/4551051/100387537-2e89e900-2fdd-11eb-9b6f-b570f92819ba.png)
Mothership:
![image](https://user-images.githubusercontent.com/4551051/100387562-38abe780-2fdd-11eb-8352-da2a105b33be.png)

Could we model the seasonal catch based on these and as Aaron mentioned, create a metric to compare with them?
Since the US fleets have two different cumulative catch rates we would have to either disaggregate the fleets or come up with a function modeling both cumulative catch curves simultaneously for the US.

**Comment by kristinmarshall-NOAA** [comment]
I'm not sure I'm totally following you here. My understanding is the OM already effectively divides the quota up by season to spread the catches out across the year.  The seasonal "quotas" were based on past patterns of fishing by country and season, as calculated by @iantaylor-NOAA, captured in the calc_catch_seas_country.R function.  These then get read in and applied in the season loop with [om$catch_props_space_season.](https://github.com/pacific-hake/pacifichakemse/blob/47198ccbf708f988e346b9eed118d503b1648f2e/R/run_season_loop_om.R#L105). 
Certainly this could be done differently or we could test different ways of doing it, e.g. with the cumulative curves suggested above. But, that wasn't what I was suggesting.

What I was trying to suggest was just to come up with a metric that would capture any instances of catch not being obtained because biomass was low in one season/area, but it was present and caught in the other area in the prior season.  I suspect it doesn't happen often in our current model with the assumptions we have about seasonal catch.  Perhaps showing that will be helpful to illustrate why this line of questioning and coming up with more complicated models of attainment is likely outside the scope of our model.  

Does that make sense?  Am I missing something in your suggestions above?

**Comment by cgrandin** [comment]
Yes, this makes more sense now. I think that if we look at this line: https://github.com/pacific-hake/pacifichakemse/blob/77a6710de8449a1014c487b161ee9ebf47867cc5/R/run_season_loop_om.R#L108

and divide it with the same minus the attainment part:
```
e_tmp_tac <- catch_space * om$catch_props_space_season[space, season] %>% pull
```
we would have the ratio of attained to TAC by country and season. We would store these by creating a new object here:
https://github.com/pacific-hake/pacifichakemse/blob/77a6710de8449a1014c487b161ee9ebf47867cc5/R/setup_mse_plot_objects.R#L36

There's a little bit more to it for the zero case (e_tmp <- 3000).

Once these ratios have been stored and the MSE run through, we can compare the previous season/other country ratio.

This part: "capture any instances of catch not being obtained because biomass was low in one season/area" can be done by adding another comparison block like this: https://github.com/pacific-hake/pacifichakemse/blob/77a6710de8449a1014c487b161ee9ebf47867cc5/R/run_season_loop_om.R#L134

**Comment by iantaylor-NOAA** [comment]
I think a challenge of this metric, in particular the "capture any instances of catch not being obtained because biomass was low in one season/area", is that it makes the most sense in the context of some maximum harvest rate. There are impacts on the economic viability of the fishery long before all the fish are gone, but even if we had all the data necessary to estimate it, that value would likely change over time and across participants in the fishery. Perhaps instead of looking for cases of catch not being obtained due to fishing in the previous season, we could measure the increase in harvest rate required to get the country-specific TAC. However, I'm not sure how best to do that without somehow repeating the seasonal population projection twice for each season, once with and once without fishing.
